matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml

```diff
@@ -127,20 +128,35 @@ $xwiki.jsx.use('PhenoTips.Widgets')##
+                  #set ($disclaimer = "$!xwiki.getDocument('PhenomeCentral.ServiceDisclaimer').getRenderedContent()")##
+                  &lt;input type="hidden" class="disclaimer" value="${escapetool.xml($disclaimer)}"&gt;##
+                  &lt;input type="hidden" class="serverName" value="$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.matchesSeletFilter.local'))"&gt;##
+                  &lt;span class="fa fa-info-circle mme-disclaimer-help" title="$escapetool.xml($services.localization.render('phenotips.myMatches.disclaimer.showDisclaimer'))"&gt;&lt;/span&gt;
+                  &lt;br&gt;
                 &lt;/label&gt;
                 &lt;div class="across-mme"&gt;&lt;input id="mme-filter" type="checkbox" checked &gt; $escapetool.xml($services.localization.render('phenotips.myMatches.filters.mme.label')) &lt;a href="http:\/\/matchmakerexchange.org" target="_blank"&gt;MME&lt;/a&gt;:&lt;/div&gt;
                 #set ($preferenceDoc = $xwiki.getDocument("XWiki.XWikiPreferences"))
                 #set ($allRemotes = $preferenceDoc.getObjects("PhenomeCentral.RemoteMatchingServiceConfiguration"))
                 #foreach ($config in $allRemotes)
                   #if ($config.getProperty('serverId').getValue() != "" &amp;&amp; $config.getProperty('humanReadableName').getValue() != "" &amp;&amp; $config.getValue('includeInUI') == 1)
-                    &lt;label&gt;&lt;input name="checkbox-server-id-filter" class="mme" type="checkbox" value="$config.getProperty('serverId').getValue()" checked&gt;
-                      &lt;div class="filter-mme-server-name"&gt;$escapetool.xml($config.getProperty('humanReadableName').getValue())&lt;span class="matches-count"&gt;&lt;/span&gt;&lt;/div&gt;&lt;br&gt;
+                    #set ($serverID = $config.getProperty('serverId').getValue())
+                    #set ($serverName = $config.getProperty('humanReadableName').getValue())
+                    #if ($serverName == "")
+                      #set ($serverName = $serverID)
+                    #end
+                    #set ($disclaimer = $config.getProperty('disclaimer').getValue())
+                    &lt;label&gt;
+                      &lt;input name="checkbox-server-id-filter" class="mme" type="checkbox" value="${escapetool.xml($serverID)}" checked&gt;
+                      &lt;input type="hidden" class="disclaimer" value="${escapetool.xml($disclaimer)}"&gt;
+                      &lt;input type="hidden" class="serverName" value="${escapetool.xml($serverName)}"&gt;
+                      &lt;div class="filter-mme-server-name"&gt;${escapetool.xml($serverName)}&lt;span class="matches-count"&gt;&lt;/span&gt;&lt;span class="fa fa-info-circle mme-disclaimer-help" title="$escapetool.xml($services.localization.render('phenotips.myMatches.disclaimer.showDisclaimer'))"&gt;&lt;/span&gt;&lt;/div&gt;
```

Comment:
Does this need to be a div? divs (block elements) inside labels are not valid markup.